### ISSUE TITLE
Missing labels on form elements

### ISSUE THREAD

**Issue by tysongach** [thoughtbot/constable#684]
Some form elements are not properly labelled, which poses UX and accessibility issues.

You can read about the accessibility implications of non-labelled forms, here: https://dequeuniversity.com/rules/axe/3.2/label?application=AxeChrome

In come cases, we are using HTML’s `placeholder` attribute as a label, but this is not what `placeholder` is designed for and it does not properly label the input. Our pal and colleague @ericwbailey wrote about [the `placeholder` attribute and its problems](https://www.smashingmagazine.com/2018/06/placeholder-attribute/).

Below is a non-exhaustive list of inputs with missing labels, as discovered by [axe](https://www.deque.com/axe/).

---

**Name input in Profile and Settings:**
<br>
![Screen Shot 2019-08-28 at 10 26 06](https://user-images.githubusercontent.com/903327/63865481-a7843600-c97f-11e9-88e1-0827c0694251.png)
<br><hr><br>
**Comment input on announcement show page:**
<br>
![Screen Shot 2019-08-28 at 10 27 22](https://user-images.githubusercontent.com/903327/63865482-a7843600-c97f-11e9-9fe6-39ee5ebceb25.png)
<br><hr><br>
**Title input on new announcement page:**
<br>
![Screen Shot 2019-08-28 at 10 28 48](https://user-images.githubusercontent.com/903327/63865483-a7843600-c97f-11e9-83bd-43c935750c79.png)
<br><hr><br>
**Body input on new announcement page:**
<br>
![Screen Shot 2019-08-28 at 10 35 06](https://user-images.githubusercontent.com/903327/63865486-a7843600-c97f-11e9-9121-26202e33c72a.png)
<br><hr><br>
**Interests input on new announcement page:**
<br>
![Screen Shot 2019-08-28 at 10 34 56](https://user-images.githubusercontent.com/903327/63865485-a7843600-c97f-11e9-9e59-555cba0dd49e.png)
<br><hr><br>
**Slack channel input:**
<br>
![Screen Shot 2019-08-28 at 10 31 47](https://user-images.githubusercontent.com/903327/63865484-a7843600-c97f-11e9-8839-5effa08dc586.png)
